lib/cybersource/api.rb

```diff
@@ -336,6 +353,15 @@ def add_required_options(kb_account_id, properties, options, context)
         ::Killbill::Plugin::ActiveMerchant::Utils.normalize_property(properties, 'ignore_cvv')
       end
 
+      def add_merchant_descriptor(transaction_type, properties, options)
+        merchant_descriptor = find_value_from_properties(properties, 'merchant_descriptor')
+        merchant_descriptor_hash = JSON.parse(merchant_descriptor) rescue nil
```

Comment:
For performance reasons, check if `merchant_descriptor` is `blank` before trying to parse it (otherwise the majority of calls will throw an unnecessary exception).